js/src/contract/item.ts

```diff
@@ -421,869 +41,283 @@ export class ItemProgram {
-            )[0]
-          : null,
-        metadataUpdateAuthority: accounts.metadataUpdateAuthority,
-        program: this.program,
-      });
-
-    const itemClassKey = (
-      await getItemPDA(accounts.itemClassMint, args.classIndex)
-    )[0];
-    const craftItemTokenAccount = (
-      await getAtaForMint(
-        accounts.craftItemTokenMint,
-        (this.program.provider as AnchorProvider).wallet.publicKey
-      )
-    )[0];
-
-    const [craftItemEscrow, _itemEscrowBump] = await getCraftItemEscrow({
-      itemClassMint: accounts.itemClassMint,
-      classIndex: args.classIndex,
-      craftIndex: args.craftItemIndex,
-      craftEscrowIndex: args.craftEscrowIndex,
-      newItemMint: args.newItemMint,
-      craftItemMint: accounts.craftItemTokenMint,
-      craftItemToken: craftItemTokenAccount,
-      payer: (this.program.provider as AnchorProvider).wallet.publicKey,
-      amountToMake: args.amountToMake,
-      amountToContributeFromThisContributor:
-        args.amountToContributeFromThisContributor,
-      componentScope: args.componentScope,
-    });
-
-    const [craftItemCounter, _craftBump] = await getCraftItemCounter({
-      itemClassMint: accounts.itemClassMint,
-      classIndex: args.classIndex,
-      craftItemIndex: args.craftItemIndex,
-      craftEscrowIndex: args.craftEscrowIndex,
-      newItemMint: args.newItemMint,
-      craftItemMint: accounts.craftItemTokenMint,
-      componentScope: args.componentScope,
-    });
-
-    const itemEscrow = (
-      await getItemEscrow({
-        itemClassMint: accounts.itemClassMint,
-        classIndex: args.classIndex,
-        craftEscrowIndex: args.craftEscrowIndex,
-        newItemMint: args.newItemMint,
-        newItemToken:
-          accounts.newItemToken ||
-          (
-            await getAtaForMint(
-              args.newItemMint,
-              args.originator ||
-                (this.program.provider as AnchorProvider).wallet.publicKey
-            )
-          )[0],
-        payer:
-          args.originator ||
-          (this.program.provider as AnchorProvider).wallet.publicKey,
-        amountToMake: args.amountToMake,
-        componentScope: args.componentScope,
-      })
-    )[0];
-
-    const craftItem = (
-      await getItemPDA(accounts.craftItemTokenMint, args.craftItemIndex)
-    )[0];
-    const craftItemObj = await this.program.account.item.fetch(craftItem);
-    const instructions = [],
-      signers = [];
-    const craftItemTransferAuthority = web3.Keypair.generate();
-
+    args: ItemInstruction.AddCraftItemToEscrowArgs,
+    accounts: ItemInstruction.AddCraftItemToEscrowAccounts,
+    additionalArgs: ItemInstruction.AddCraftItemToEscrowAdditionalArgs,
+    options?: SendOptions
+  ): Promise<{
+    txid: string;
+    slot: number;
+  }> {
+    const signers = [];
+    const craftItemTransferAuthority = web3.Keypair.generate(); // Pulling this up out of the instruction - is that ok?
```

Comment:
@long-banana Also not sure here, previously this could not be specified by the user, but with the instruction separate, this could be overridden if you use the instruction directly. I pulled it out to pass into the send call below.

Reply:
How about let's make this the default in the instruction if it isn't specified by the caller? Allow it to be overridden with the instruction being used directly, but don't force it to be supplied. That simplifies the Program instance to just use the defaults. Which I think is a good pattern to have for these basic classes that extend Program.

Reply:
@long-banana We need to pass this into send below as one of the signers: https://github.com/raindrops-protocol/raindrops/pull/17/files#diff-2ddcd821bbf8cba92935c6e2e096c56617ece2b20dd4da9cf1f7a866b85db4e9R163
So the program needs to know about it somehow

Reply:
I see and realize that now. The way you have it sounds good to me then. It makes sense to force someone to pass it into the instruction in that case. Otherwise, not going to be able to sign it like you said.